### ISSUE TITLE
Option to return raw list of Vars instead of Dataset in open() routines

### ISSUE THREAD

**Issue by neishm** [pygeode/pygeode#57]
Original [issue 57](https://code.google.com/p/pygeode/issues/detail?id=57) created by neishm on 2014-05-22T17:29:04.000Z:

For my current work, I'm dealing with files which could have different sets of levels, latitudes, and longitudes for different variables.  These axes all get mapped to unique names like lat01, lat02, lon01, lon02 when the data is wrapped as a Dataset at the end of the open() routine.  Normally, I only want to look at one of the variables at a time anyway, so I don't need this extra wrapping.  It's one extra layer of stack trace I have to sort through when there's problems.  It could also cause problems if the order of the variables (and thus unique axis names) changes from file to file, confusing the hell out of the multifile interface.

What I've done to work around this issue is add a &quot;raw_list&quot; flag to my format's opener.  If raw_list=True, it skips the final Dataset wrapper, and returns the Vars in a simple list.  Other record-based formats (like CCCma files and GRIB) could benefit from a similar option.  I was wondering if it would make sense to add such a flag to all open() methods, to skip the Dataset wrapper and maybe save a small amount of overhead in the process?

**Comment by aph42** [comment]
Yes, I think this is a good option - I can see this being useful for similar reasons. Since it should be a quick fix, 1.1?

**Comment by neishm** [comment]
Sure, sounds good to me.

**Comment by neishm** [comment]
Should this `raw_list` option be compatible with the other options (`value_override`, `dimtypes`, `namemap`, `varlist`)?  Because these options, as implemented by `finalize_open`, are built around the assumption that we've already constructed a `Dataset` object (which defeats the whole purpose of `raw_list`).

How about making this a mutually exclusive option for now, and raise a `ValueError` or `KeyError` if it's mixed with the other arguments?

**Comment by neishm** [comment]
Since this isn't as quick a fix as I originally thought, I'm going to defer this to a future release.
